macros/src/codegen/shared_resources.rs

```diff
@@ -124,18 +119,31 @@ pub fn codegen(
             None
```

Comment:
What to do when this happens? Panic? If this just "falls through" we have have UB.

Reply:
We could reject RTIC applications on armv6 that uses shared lock free resources tied to exceptions (the exceptions cannot be easily masked by the NVIC). This would be a very rare case though, e.g., using SysTick as a task with a shared resource, the monotonic implementation using SysStick should be fine as it does not use any shared resource (at least not the current implementation). A short summary of the v6 exceptions, and usability as generic tasks.

1 | Reset, not usable
2 | NMI, not usable
3 | HardFault, not usable
4-10 | Reserved, not usable
11 | SVCall, ... hmm perhaps, but never seen it....
12-13 | Reserved
14 | PendSV, ... hmm perhaps, but never seen it
15 | SysTick, ... seems like the only usable case

Reply:
The latest fix proposal should take care of that, our current way to perform tests on v6 vs v7 is not covering all cases though, and the whole test approach needs to be revisited at some point. I think however we can live with this atm.